src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterKeyValuePair.cs

```diff
@@ -94,7 +94,7 @@ private T ReadProperty<T>(ref Utf8JsonReader reader, Type typeToConvert, JsonSer
 
             // Attempt to use existing converter first before re-entering through JsonSerializer.Deserialize().
             // The default converter for objects does not parse null objects as null, so it is not used here.
-            if (typeToConvert != typeof(object) && (options.GetConverter(typeToConvert) is JsonConverter<T> keyConverter))
+            if (typeToConvert != typeof(object) && (options?.GetConverter(typeToConvert) is JsonConverter<T> keyConverter))
```

Comment:
Did we discover this while testing?

Are these the only instances where null options were being passed in and could result in a nullref?

Looks like we have a few converters that use options and don't check for null. What about some of the `JsonDictionaryConverter`?
https://github.com/dotnet/corefx/blob/04bd3fa5c2c7d379c22b805ad840bc28423d5242/src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultDerivedDictionaryConverter.cs#L11

@layomia, @steveharter - do we need additional guards against null options?

Reply:
> Did we discover this while testing?

Yes
> Are these the only instances where null options were being passed in and could result in a nullref?

This is tests only covers converters implemented JsonConverter interface and available through JsonSerializerOptions.Getconverter(type) method. Within that range yes this is only case were resulting NRE 
  
> Looks like we have a few converters that use options and don't check for null. What about some of the `JsonDictionaryConverter`?

This PR didn't include any Dictionary or Collection converters as they are not implemented JsonConverter, i have added custom dictionary converter test though https://github.com/dotnet/corefx/pull/39750#discussion_r307537607